docs/rfcs/XXXX_storage_backends.md

```diff
@@ -0,0 +1,248 @@
+
+### Application Bucket
+
+It's proposed that Dewrangle maintain a single bucket in which it may stash data. This bucket will be intended to contain anything the Dewrangle may need to store such as configuration files, user avatars, and similar. Most notably, though, it will contain uploaded documents for each study. In this way, the single application bucket will deprecate the need for each study bucket to contain its uploads. However, it is also recognized that perhaps some organizations may wish their uploaded documents to reside within their own resource pool. For this case, we will allow the document bucket to be specified within a different account.
+The default solution for a new organization (and the existing Kids First organization) will be to store all documents within the application bucket. These documents will be kept as outlined above in the `New Layout` where the `studies/` prefix will be at the root of the bucket.
+
+### New Layout
+```
+studies/
+  SD_00000000/
+    source/
+    uploads/
+    harmonized/
+```
+
+## External Storage Backends
+
+![enter image description here](https://i.ibb.co/PTPMLnf/storage-layout.png)
+Under circumstances that an organization requires all of their uploads or data files to be under their control, an organization administrator may specify credentials and details for a bucket of their choosing in a different account.
+
+### Storage Backends
+
+A Storage Backend is a simple entity that belongs to an organization and specifies IAM user credentials, a bucket that the IAM user has access to, and a prefix within that bucket. This concept will allow Dewrangle to access buckets across different accounts for the purpose of storing or evaluating files.
+
+### Configuration
+To configure a storage backend, an organization administrator will be required to provide details about the AWS account where their target bucket resides and a user who may access it. An organization administrator will be able to configure this by navigating to the *Organization Settings* screen and choosing *Storages*. This will allow the administrator to add a new storage by first asking the user to create a new IAM user in their AWS account and assigning a specified policy. After the user has done this, they will be required to provide the AWS access and secret key pair so that Dewrangle may access the bucket. The final step will be to allow the user to specify the bucket and what prefix under it they would like this storage backend to use as the root. After confirming their choices, Dewrangle will attempt to access the bucket with the provided credentials and save the new storage backend if successful.
```

Comment:
i'm not sure what `The final step will be to allow the user to specify the bucket and what prefix under it they would like this storage backend to use as the root.` means

Reply:
Does the user flow where the user is asked to provide bucket info make more sense? Prefixes are like folders in S3, so the user may optionally provide a prefix inside the bucket where all the data where reside. It's like git looking for `.git` and only being concerned about whatever files lie below it in the file system.

Reply:
yeah that makes sense